src/sidebar/services/api.js

```diff
@@ -82,15 +82,23 @@ function isRouteMetadata(link) {
  * @param {string} route - Dot-separated path of route in `routeMap`
  */
 function findRouteMetadata(routeMap, route) {
-  /** @type {RouteMap|RouteMetadata} */
+  /** @type {RouteMap} */
   let cursor = routeMap;
-  for (let segment of route.split('.')) {
-    cursor = cursor[segment];
```

Comment:
This line of code was technically incorrect if we find a `RouteMetadata` (object with `url` and `method` properties) before we have reached the end of the dotted-path specified by `route`.

For example, if `routeMap` looked like this:

```js
{
  profile: {
    read: { method: 'GET', url: '...' }
  }
}
```

And `route` is `profile.read.something`, then when we reach the `read` segment this function we'll find a `RouteMetadata` object unexpectedly. In this case the function should return `null`.

The previous code would give the same result in this scenario because it would look for a `something` key in the `{ method, url, ... }` object and not find it. Nevertheless, the new code more directly handles this possibility.